sdlib/d/gc/tcache.d

```diff
@@ -570,3 +571,31 @@ unittest extend {
 	// Extend by size zero still works, though:
 	assert(threadCache.extend(p1[0 .. 16384], 0));
 }
+
+unittest arraySpill {
+	void setUsed(void* ptr, size_t usedCapacity) {
+		assert(ptr !is null);
+		auto pd = threadCache.getPageDescriptor(ptr);
+		assert(pd.extent !is null);
+		assert(pd.extent.isLarge());
+		pd.extent.setUsedCapacity(usedCapacity);
+	}
+
+	// Verify that zero-length interstitial slice does not spill into next alloc:
+	void testAdjacentSpill(size_t arrSize, size_t a1Cap, size_t a2Cap) {
+		auto array1 = threadCache.alloc(arrSize, false);
+		auto array2 = threadCache.alloc(arrSize, false);
+		assert(array2 is array1 + arrSize);
+		setUsed(array1, a1Cap);
+		setUsed(array2, a2Cap);
+		assert(threadCache.getCapacity(array1[arrSize .. arrSize]) == 0);
```

Comment:
You might want to check that you get the result you expect as per capacity, EXCEPT when the slice is 0 and the capacity is zero.

This is not reflected in the test case. This test always expecting the same result, so clearly not testing the code. That damn thing could always return 0, instead of returning 0 in a specific situation, that this would still pass.